### PR TITLE
:bar_chart: adding regions to remittance data

### PR DESCRIPTION
**Pull request by antea04**
(for owid/owid-issues#1699)

Adding regional averages to remittance costs in the WDI and adding remittances over regions. 

Average remittance costs are calculated only including countries where both volume and cost of remittances is available.

- indicators ([sending costs](http://staging-site-remittance-regions/admin/variables/905930), [receiving costs](http://staging-site-remittance-regions/admin/variables/905929)) on staging
- <s>since not everywhere costs and volumes are available, some regional averages are only a few countries. I think it is still useful for the overall trend, but open to feedback on how to deal with this </s> 
_edit:_ I've now changed the code to only include averages when more than 75% of the remittance volume in an area is covered by cost data.

### CODE REVIEW

File: etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py [hunk 1]
```diff
@@ -77,6 +83,73 @@ def run(dest_dir: str) -> None:
...
+    Add regions to remittance data.
+    """
+
+    tb = tb.reset_index()
+
+    # create a copy so other indicators are not affected
+    regions_tb = tb.copy()
+
+    # create new columns for total remittances (only for countries where remittance cost is available)
+    regions_tb["total_received_remittances"] = regions_tb["bx_trf_pwkr_cd_dt"].where(
```
Comment (paarriagadap) [inline etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py:97]: Not sure why this step is needed. Why not just multiplying si_rmt_cost_ib_zs * bx_trf_pwkr_cd_dt and si_rmt_cost_ob_zs * bm_trf_pwkr_cd_dt direclty? The nulls would be generated automatically

File: etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py [hunk 2]
```diff
@@ -77,6 +83,73 @@ def run(dest_dir: str) -> None:
...
+        "bm_trf_pwkr_cd_dt": "sum",
+    }
+
+    # add regions to table
+    regions_tb = geo.add_regions_to_table(
+        regions_tb,
+        ds_regions=ds_regions,
+        ds_income_groups=ds_income_groups,
+        aggregations=agg,
+        min_num_values_per_year=1,
```
Comment (paarriagadap) [inline etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py:127]: For me this is a very permissive restriction, that can maybe not useful fore regional analysis. I am more inclined to using frac_allowed_nans_per_year

**Review by paarriagadap — APPROVED** [review]
Hi Tuna! Here are my comments. In general, I am not a big fan of regional aggregations, and especially in cases where the data is scarce. I would say if there are not enough countries in a continent we should not aggregate, because the resulting number can be misleading.

I think we can create a good DI by showing this data for several countries in different regions and it would work just fine.

If Simon and you really want the aggregations, I think defining a fraction of nan threshold per region is adequate.

File: dag/main.yml [hunk 3]
```diff
@@ -653,6 +653,8 @@ steps:
     - snapshot://worldbank_wdi/2024-05-20/wdi.zip
     - data://meadow/worldbank_wdi/2024-05-20/wdi
     - data://garden/demography/2024-07-15/population
+    - data://garden/regions/2023-01-01/regions
+    - data://garden/wb/2024-03-11/income_groups
```
Comment (paarriagadap) [inline dag/main.yml:657]: There is a newer version of the income_groups dataset: 2024-07-29

File: etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py (hunk 1 above)
Comment (antea04) [reply etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py:104]: I need this column to calculate "total remittance volume covered by cost data" to calculate the right average and (now) cover the coverage of the regions.

File: etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py (hunk 2 above)
Comment (antea04) [reply etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py:135]: I've added a filter for the regional average calculation afterwards - only include the average if more than 75% of transaction volume has an average cost associated.

**Comment by antea04** [comment]
@paarriagadap Hey Pablo! Thank you so much for your review and weighing in on the data coverage issue. I totally agree with your assessment of this.

In this case, just taking the number/share of NAs for countries in each area would be a bit misleading, since often very few countries in one area send/ receive the most remittances (e.g USA/ Canada in NA, Australia in Oceania). To combat this I've now changed the code to only include averages when more than 75% of the remittance volume in an area is covered by cost data. (I think this a reasonable large coverage).

Let me know what you think - if you give me a 👍 I'll go ahead and merge!

**Review by paarriagadap — APPROVED** [review]
Hi! I think it's a better approach. I would still let Ed know if that's fine with him as well, to have a second opinion

**Comment by edomt** [comment]
Sorry for the delay @antea04 — I finally took at this now. The logic makes sense to me.

Looking at http://staging-site-remittance-regions/admin/variables/905929, this 75% threshold actually gives us data points for all years between 2011 and 2020 for the "important" regions (Africa, Asia, and South America). I'm curious: what happens if you pick a more aggressive threshold, say, 90%? How many of these data points are still present?

image

**Comment by antea04** [comment]
Hi @edomt - no worries, thanks for taking a look!

Yeah, I thought 75% is a pretty solid threshold while still keeping many interesting data points around. The "important" receiving regions still have solid data for most years with a higher threshold (say 90%), but for regions from which remittances are sent it gets pretty sparse at ~80%.

To satisfy your curiosity, **here the full breakdown.** I'm open to change the threshold if you think that makes sense! 

----
### with threshold 0.75 following regions and years are still included
#### Sending remittances:
Asia: [2016 2017 2020]
Europe: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
High-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
North America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Oceania: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

#### Receiving remittances:
Africa: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Asia: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Low-income countries: [2012 2013 2014 2015 2016 2017 2018 2019 2020]
Lower-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
North America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
South America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Upper-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

### with threshold 0.8 following regions and years are still included
#### Sending remittances:
Asia: [2016]
Europe: [2012 2013 2014 2015 2016]
High-income countries: [2012 2013 2016 2017 2018 2019 2020]
North America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Oceania: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

#### Receiving remittances:
Africa: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Asia: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Low-income countries: [2013 2014 2016 2017 2018 2019 2020]
Lower-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
North America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
South America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Upper-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

### with threshold 0.85 following regions and years are still included
#### Sending remittances:
High-income countries: [2016 2017 2018 2019 2020]
North America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Oceania: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
#### Receiving remittances:
Africa: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Asia: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Low-income countries: [2016 2017 2018 2019 2020]
Lower-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
North America: [2016 2017 2018 2019 2020]
South America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Upper-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

### with threshold 0.9 following regions and years are still included
#### Sending remittances:
High-income countries: [2016]
North America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Oceania: [2012 2013 2014 2015 2016 2017 2018 2019 2020]

#### Receiving remittances:
Africa: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Asia: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Low-income countries: [2016 2017 2018 2019 2020]
Lower-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
South America: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
Upper-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

### with threshold 0.95 following regions and years are still included
#### Sending remittances:
North America: [2011 2012 2013]
Oceania: [2018 2019 2020]
#### Receiving remittances:
Africa: [2012 2013 2014 2015 2016 2017 2018 2019 2020]
Lower-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]
South America: [2017 2018 2019 2020]
Upper-middle-income countries: [2011 2012 2013 2014 2015 2016 2017 2018 2019 2020]

**Review by edomt — APPROVED** [review]
Thanks for satisfying my curiosity! Though looking at the results, I agree with you that 75% seems like a good threshold.

We're good to go!

**Pull request merged**